earth2mip/schema.py

```diff
@@ -40,6 +41,30 @@ def shape(self):
         else:
             raise ValueError(f"Unknown grid {self}")
 
+    @property
```

Comment:
Nice! I have nearly identical logic in my graphcast branch.

Ultimately, I would like to make a more flexible Grid type i.e. `RegularLatLon` matching this API. The use of enum is mostly legacy, now that the plugin mechanism allows setting time_loop.grid at runtime.